[kafka] Expose resolve_canonical_bootstrap_servers_only

gebn opened the pull request:
**Description:** The latest release of IBM/sarama includes the `resolve_canonical_bootstrap_servers_only` setting for `client.dns.lookup`. In a GSSAPI context, this enables replacing a list of brokers with a single name that resolves to all of the underlying IPs. Previously this would fail with a Kerberos error. This PR exposes the new option in collector config.

**Testing:** Added new cases to check the option is set correctly.

**Documentation:** Added a description of the parameter to the relevant READMEs.

Commit:
Expose Kafka resolve_canonical_bootstrap_servers_only

mx-psi reviewed (commented):
@pavolloffay @MovieStoreGuy can you take a look as codeowners?

mx-psi commented:
@pavolloffay @MovieStoreGuy ping

gebn commented:
@pavolloffay @MovieStoreGuy ping, please let me know of any amendments!

gebn commented:
Sarama has since been upgraded automatically in `main`, so I've rebased and removed that commit - the diff is now much more related to only the intent. I'm still keen to get this merged to simplify our Kafka configuration - please let me know what needs to be done. Build failures appear unrelated to the change.

@pavolloffay @MovieStoreGuy @mx-psi

gebn commented:
Sarama has since been upgraded automatically in `main`, so I've rebased and removed that commit - the diff is now much more related to only the intent. I'm still keen to get this merged to simplify our Kafka configuration - please let me know what needs to be done. Build failures appear unrelated to the change.

@dmitryax @MovieStoreGuy @mx-psi

gebn commented:
@mx-psi Apologies, there is a trivial import error, I'll fix this this evening - unable to at work.

gebn commented:
All sorted @mx-psi
